source/MRMesh/MRVoxelsLoad.cpp

```diff
@@ -471,66 +470,90 @@ std::shared_ptr<ObjectVoxels> loadDCMFolder( const std::filesystem::path& path,
         data.max = std::max( sliceRes.max, data.max );
     }
 
-    ObjectVoxels voxels;
-    voxels.construct( data, [&]( float proc )
-    {
-        if ( cb )
-            cb( 0.7f + 0.3f*proc );
-        return true;
-    } );
+    if ( cb )
+        newCb = [&cb] ( float f ) { return cb( 0.5f + 0.5f * f ); };
+    LoadDCMResult res;
+    res.vdbVolume = simpleVolumeToVdbVolume( data, newCb );
     if ( firstRes.seriesDescription.empty() )
-        voxels.setName( utf8string( files.front().stem() ) );
+        res.name = utf8string( files.front().stem() );
     else
-        voxels.setName( firstRes.seriesDescription );
-    return std::make_shared<ObjectVoxels>( std::move( voxels ) );
+        res.name = firstRes.seriesDescription;
+    return res;
 }
 
-std::vector<std::shared_ptr<ObjectVoxels>> loadDCMFolderTree( const std::filesystem::path& path, unsigned maxNumThreads, const ProgressCallback& cb )
+std::vector<tl::expected<LoadDCMResult, std::string>> loadDCMFolderTree( const std::filesystem::path& path, unsigned maxNumThreads, ProgressCallback cb )
 {
     MR_TIMER;
-    std::vector<std::shared_ptr<ObjectVoxels>> res;
+    std::vector<tl::expected<LoadDCMResult, std::string>> res;
+    bool anySuccessLoad{ false };
     auto tryLoadDir = [&]( const std::filesystem::path& dir )
     {
-        if ( auto obj = loadDCMFolder( dir, maxNumThreads, cb ) )
-            res.push_back( std::move( obj ) );
+        auto loadRes = loadDCMFolder( dir, maxNumThreads, cb );
+        if ( loadRes.has_value() )
+        {
+            res.push_back( *loadRes );
+            anySuccessLoad = true;
+        }
+        else
+        {
+            const std::string str = "loadDCMFolder: there is no dcm file in folder:";
+            if ( anySuccessLoad && loadRes.error().substr( 0, str.size() ) == str )
+                return true;
+            res.push_back( loadRes );
+            if ( loadRes.error() == "Loading cancelled" )
```

Comment:
cancelled -> canceled (and in other places)

Reply:
fixed